syntax = "proto2";

message PublisherKeyCmdBuf {
	oneof PublisherKeyCmdType {
		PublisherKeyRequest key_request = 1;
		PublisherDagRequest dag_request = 2;
		PublisherSignRequest sign_request = 3;
		PublisherVerifyRequest verify_request = 4;
	}
}

message PublisherKeyResponseBuf {
	oneof PublisherKeyResponseType {
		PublisherKeyResponse key_response = 1;
		PublisherDagResponse dag_response = 2;
		PublisherSignResponse sign_response = 3;
		PublisherVerifyResponse verify_response = 4;
	}
}

message PublisherKeyRequest {
	required string publisher_name = 1;
}

message PublisherKeyResponse {
	required bool success = 1;
	optional string publisher_key = 2;
}

message PublisherDagRequest {
	required string publisher_name = 1;
}

message PublisherDagResponse {
	required bool success = 1;
	optional string cert_dag = 2;
}
message PublisherSignRequest {
	required string publisher_name = 1;
	required bytes digest = 2;
}

message PublisherSignResponse {
	required bool success = 1;
	optional bytes signature = 2;
}

message PublisherVerifyRequest {
	required string publisher_name = 1;
	required bytes signature = 2;
	required bytes digest = 3;
}

message PublisherVerifyResponse {
	required bool success = 1;
}
